quickwit-serve/src/lib.rs

```diff
@@ -21,77 +21,104 @@ mod args;
 mod counters;
 mod error;
 mod format;
+
 mod grpc;
+mod rest;
 
 mod cluster_api;
 mod health_check_api;
 mod search_api;
 
-mod rest;
-
+use std::collections::HashSet;
 use std::sync::Arc;
 
+use anyhow::bail;
 use format::Format;
-use quickwit_cluster::cluster::Cluster;
-use quickwit_cluster::service::ClusterServiceImpl;
-use quickwit_config::{QuickwitConfig, SEARCHER_CONFIG_INSTANCE};
-use quickwit_metastore::Metastore;
-use quickwit_search::{ClusterClient, SearchClientPool, SearchService, SearchServiceImpl};
+use quickwit_cluster::ClusterService;
+use quickwit_config::QuickwitConfig;
+use quickwit_indexing::actors::IndexingServerClient;
+use quickwit_indexing::start_indexer_service;
+use quickwit_metastore::quickwit_metastore_uri_resolver;
+use quickwit_search::{start_searcher_service, SearchService};
 use quickwit_storage::quickwit_storage_uri_resolver;
-use tracing::{debug, info};
 
 pub use crate::args::ServeArgs;
-use crate::cluster_api::GrpcClusterAdapter;
 pub use crate::counters::COUNTERS;
-use crate::grpc::start_grpc_service;
 #[cfg(test)]
 use crate::rest::recover_fn;
-use crate::rest::start_rest_service;
-use crate::search_api::GrpcSearchAdapter;
 
-/// Starts a search node, aka a `searcher`.
-pub async fn run_searcher(
-    quickwit_config: QuickwitConfig,
-    metastore: Arc<dyn Metastore>,
-) -> anyhow::Result<()> {
-    SEARCHER_CONFIG_INSTANCE
-        .set(quickwit_config.searcher_config.clone())
-        .expect("could not set searcher config in global once cell");
-    let cluster = Arc::new(Cluster::new(
-        quickwit_config.node_id.clone(),
-        quickwit_config.gossip_socket_addr()?,
-    )?);
-    for seed_socket_addr in quickwit_config.seed_socket_addrs()? {
-        // If the peer seed address is specified,
-        // it joins the cluster in which that node participates.
-        debug!(peer_seed_addr = %seed_socket_addr, "Add peer seed node.");
-        cluster.add_peer_node(seed_socket_addr).await;
+#[derive(Debug, PartialEq, Eq, Copy, Hash, Clone)]
+pub enum QService {
```

Comment:
I would probably choose `QuickwitService` as we tend to use struct with `Quickwit` prefix.